## Project Start and Insert Dataset

In [77]:
# Begin Project Code
import numpy as np
import pandas as pd  

#import dataset into document and ensure data has loaded properly
data = pd.read_csv(r"C:\Users\wills\OneDrive\uh\Coursea Analytics\CSV\tiktok_dataset (1).csv")
data.head()

# claim_status    video_id  video_duration_sec  \
0  1        claim  7017666017                  59   
1  2        claim  4014381136                  32   
2  3        claim  9859838091                  31   
3  4        claim  1866847991                  25   
4  5        claim  7105231098                  19   

                            video_transcription_text verified_status  \
0  someone shared with me that drone deliveries a...    not verified   
1  someone shared with me that there are more mic...    not verified   
2  someone shared with me that american industria...    not verified   
3  someone shared with me that the metro of st. p...    not verified   
4  someone shared with me that the number of busi...    not verified   

  author_ban_status  video_view_count  video_like_count  video_share_count  \
0      under review          343296.0           19425.0              241.0   
1            active          140877.0           77355.0            19034.0   
2            active          902185.0           97690.0             2858.0   
3            active          437506.0          239954.0            34812.0   
4            active           56167.0           34987.0             4110.0   

   video_download_count  video_comment_count  
0                   1.0                  0.0  
1                1161.0                684.0  
2                 833.0                329.0  
3                1234.0                584.0  
4                 547.0                152.0

In [78]:
#dataset informational statistics 
data.info()
data.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19382 entries, 0 to 19381
Data columns (total 12 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   #                         19382 non-null  int64  
 1   claim_status              19084 non-null  object 
 2   video_id                  19382 non-null  int64  
 3   video_duration_sec        19382 non-null  int64  
 4   video_transcription_text  19084 non-null  object 
 5   verified_status           19382 non-null  object 
 6   author_ban_status         19382 non-null  object 
 7   video_view_count          19084 non-null  float64
 8   video_like_count          19084 non-null  float64
 9   video_share_count         19084 non-null  float64
 10  video_download_count      19084 non-null  float64
 11  video_comment_count       19084 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 1.8+ MB


#      video_id  video_duration_sec  video_view_count  \
count  19382.000000  1.938200e+04        19382.000000      19084.000000   
mean    9691.500000  5.627454e+09           32.421732     254708.558688   
std     5595.245794  2.536440e+09           16.229967     322893.280814   
min        1.000000  1.234959e+09            5.000000         20.000000   
25%     4846.250000  3.430417e+09           18.000000       4942.500000   
50%     9691.500000  5.618664e+09           32.000000       9954.500000   
75%    14536.750000  7.843960e+09           47.000000     504327.000000   
max    19382.000000  9.999873e+09           60.000000     999817.000000   

       video_like_count  video_share_count  video_download_count  \
count      19084.000000       19084.000000          19084.000000   
mean       84304.636030       16735.248323           1049.429627   
std       133420.546814       32036.174350           2004.299894   
min            0.000000           0.000000              0.000000   
25%          810.750000         115.000000              7.000000   
50%         3403.500000         717.000000             46.000000   
75%       125020.000000       18222.000000           1156.250000   
max       657830.000000      256130.000000          14994.000000   

       video_comment_count  
count         19084.000000  
mean            349.312146  
std             799.638865  
min               0.000000  
25%               1.000000  
50%               9.000000  
75%             292.000000  
max            9599.000000

## Data Exploration - Understand the data - Investigate the variables

In [70]:
#defines the different reponses in the dataset's "claim_status" column 
uniqueclaimscategories = data['claim_status'].unique()
print(uniqueclaimscategories)

#shows how many counts of each dataset of claims and displays the data type for each
data['claim_status'].value_counts()

['claim' 'opinion' nan]


claim_status
claim      9608
opinion    9476
Name: count, dtype: int64

**Notes:** The values shown show a even distribution of both "claim" and "opinion" categories within the claim_status column

In [71]:
# What is the average view count of videos with "claim" status?
# What is the average view count of videos with "opinion" status?
### YOUR CODE HERE ###

statusclaimviewcount = (data['claim_status'] == "claim") | (data['claim_status'] =="opinion") #groups the claim status for both claim and opinion into statusclaimviewcount with a mask

data[statusclaimviewcount].groupby('claim_status').agg({'video_view_count': ['mean','median']}) #groups the claims by claim_status, then displays the average and median of the claims together


video_view_count          
                         mean    median
claim_status                           
claim           501029.452748  501555.0
opinion           4956.432250    4953.0

**Notes:** The claim category has significantly higher mean and median claim status's than the opinion category

In [79]:
# Get counts for each group combination of claim status and author ban status
### YOUR CODE HERE ###
data.groupby(["claim_status",'author_ban_status']).count()
data.groupby(["claim_status",'author_ban_status']).count()["video_id"]


claim_status  author_ban_status
claim         active               6566
              banned               1439
              under review         1603
opinion       active               8817
              banned                196
              under review          463
Name: video_id, dtype: int64

**Notes:** The author banned 

In [73]:
# What's the median video share count of each author ban status?
### YOUR CODE HERE ###
data.groupby('author_ban_status')['video_share_count'].agg("median")

author_ban_status
active            437.0
banned          14468.0
under review     9444.0
Name: video_share_count, dtype: float64

**Notes:** Banned features significantly more video share count than active users

In [74]:
#creates a table showing video_view_count, like_count and share_count by "author_ban_status"
data.groupby('author_ban_status').agg({'video_view_count':  ("count","mean","median"),
                                       'video_like_count':  ("count","mean","median"),
                                       'video_share_count': ("count","mean","median")})

video_view_count                          video_like_count  \
                             count           mean    median            count   
author_ban_status                                                              
active                       15383  215927.039524    8616.0            15383   
banned                        1635  445845.439144  448201.0             1635   
under review                  2066  392204.836399  365245.5             2066   

                                           video_share_count                \
                            mean    median             count          mean   
author_ban_status                                                            
active              71036.533836    2222.0             15383  14111.466164   
banned             153017.236697  105573.0              1635  29998.942508   
under review       128718.050339   71204.5              2066  25774.696999   

                            
                    median  
author_ban_status           
active               437.0  
banned             14468.0  
under review        9444.0

**Notes:** Banned users have significantly more views, likes and shares than active users

In [75]:
#creates "likes_per_view" column in the dataset
data["likes_per_view"] = (data['video_like_count'] / data["video_view_count"])

#creates "comments_per_view" column in the dataset
data["comments_per_view"] = (data["video_comment_count"]) / data["video_view_count"]

#creates "shares_per_view" column in the dataset
data['shares_per_view'] = (data["video_share_count"]) / data["video_view_count"]

#prints dataframe to ensure new categories are inserted in to the dataset
data.head(3)



# claim_status    video_id  video_duration_sec  \
0  1        claim  7017666017                  59   
1  2        claim  4014381136                  32   
2  3        claim  9859838091                  31   

                            video_transcription_text verified_status  \
0  someone shared with me that drone deliveries a...    not verified   
1  someone shared with me that there are more mic...    not verified   
2  someone shared with me that american industria...    not verified   

  author_ban_status  video_view_count  video_like_count  video_share_count  \
0      under review          343296.0           19425.0              241.0   
1            active          140877.0           77355.0            19034.0   
2            active          902185.0           97690.0             2858.0   

   video_download_count  video_comment_count  claimviewcount  likes_per_view  \
0                   1.0                  0.0            True        0.056584   
1                1161.0                684.0            True        0.549096   
2                 833.0                329.0            True        0.108282   

   comments_per_view  shares_per_view  
0           0.000000         0.000702  
1           0.004855         0.135111  
2           0.000365         0.003168

In [76]:
#groups both claim status and author ban status from the dataframe and then aggregates the categories "likes_per_view", "comments_per_view","shares_per_view" and displays each categories: 
# Mean, Median, and Count
data.groupby(['claim_status','author_ban_status']).agg({"likes_per_view": ("count","mean","median"),
                                                        "comments_per_view": ("count","mean","median"),
                                                        "shares_per_view": ("count","mean","median")})

likes_per_view                      \
                                        count      mean    median   
claim_status author_ban_status                                      
claim        active                      6566  0.329542  0.326538   
             banned                      1439  0.345071  0.358909   
             under review                1603  0.327997  0.320867   
opinion      active                      8817  0.219744  0.218330   
             banned                       196  0.206868  0.198483   
             under review                 463  0.226394  0.228051   

                               comments_per_view                      \
                                           count      mean    median   
claim_status author_ban_status                                         
claim        active                         6566  0.001393  0.000776   
             banned                         1439  0.001377  0.000746   
             under review                   1603  0.001367  0.000789   
opinion      active                         8817  0.000517  0.000252   
             banned                          196  0.000434  0.000193   
             under review                    463  0.000536  0.000293   

                               shares_per_view                      
                                         count      mean    median  
claim_status author_ban_status                                      
claim        active                       6566  0.065456  0.049279  
             banned                       1439  0.067893  0.051606  
             under review                 1603  0.065733  0.049967  
opinion      active                       8817  0.043729  0.032405  
             banned                        196  0.040531  0.030728  
             under review                  463  0.044472  0.035027

**Notes** After making new categories, we can see that "opinion" offers a higher active count, while its counts in likes, comments, and shares are signficantly less than the banned users in claim

## Summary Questions

- What percentage of the data is comprised of claims and what percentage is comprised of opinions?
    -  The Percentage of data is split between claims and opinion is 51% for claims(9,608) and 49% for opinions
- What factors correlate with a video's claim status?
    -  Videos with banned/ under review users have a higher claim classification in the dataset
- What factors correlate with a video's engagement level?
    -  videos with a higher count of likes, shares and comments (per view) are more likely to be Claim Videos. In general, videos with banned authors have higher engagement in videos than active authors. 